river/tree/__init__.py

```diff
@@ -7,11 +9,13 @@
 from .label_combination_hoeffding_tree import LabelCombinationHoeffdingTreeClassifier
 
 __all__ = [
+    "splitter",
+    "HoeffdingTree",
```

Comment:
Not sure we want to expose this base class, right?

Reply:
It's kind of tricky. It has some important documentation, that's the only reason it's public right now. I'm not sure this is the best approach, but it was a way to avoid repetition.

Reply:
I'm not 100% but I think we can play with the `__doc__` attribute of classes. So you could have some string that contains the doc, and just add it the `__doc__` of the class you want to add the documentation to. I've never done it but I've seen it done in (serious) projects :)

Reply:
That's a good point. Keep in mind, however, that we do have some other abstract classes that are exposed in the documentation. For example, `optim.Optimizer`. I am not sure whether or not this was intended.

> On the other hand, it is unlikely that people will find or search for the documentation in a different class.

That is a valid point. Currently, we point to `tree.HoeffdingTree` from the other trees. Something like: "for more information check `tree.HoeffdingTree`".

Again, I would rather keep the plethora of somewhat obscure (yet so useful) parameters documented in the base class than exposing them everywhere and confusing the user with too many choices to pick from. A page in the documentation could do the work, but I find it strange to document parameters outside of a docstring. If we opt for the first option and replicate the parameter information everywhere, we can follow Max's suggestion.

[`matplotlib`](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.plot.html) also relies a lot on `**kwargs`. Maybe we could provide some pieces of information about what the user can do and then point to the base class. Hence, we balance the amount of the "data overhead" for the user.

Reply:
@MaxHalford, would it be possible to add this portion of the documentation at the "index" page of the tree module? We do not have that now, but it could be a viable solution.

I am also convinced that `HoeffdingTree` should not be public.

Reply:
Yes we can and should documentation in the `__init__.py` file if that's what you mean.

Reply:
Sounds like a plan. Probably I will not use the same notation, (the `Parameters` section, for instance) but it will do the work!